src/handlers/secret.rs

```diff
@@ -0,0 +1,182 @@
+use log::debug;
+use rocket::http::Status;
+use rocket::response::status;
+use rocket::serde::json::Json;
+use rocket::{Response, State};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use serde_json::Value;
+use uuid::Uuid;
+use crate::rocket::futures::TryStreamExt;
+
+use crate::Storage;
+use crate::{
+    error::AppError,
+    storage::{
+        secret::{NewSecret, SecretEntity},
+        storage::AttemptCountRule,
+    },
+};
+
+#[derive(Serialize)]
+pub struct AllSecretsResponse {
+    secrets: Vec<Secret>,
+}
+
+#[derive(Clone, Deserialize)]
+pub struct GuessSecret {
+    guess: String,
+    username: String,
+}
+
+#[derive(Clone, Debug, Serialize)]
```

Comment:
We should use the macro `#[serde(skip_serializing_if = "Option::is_none")]` in the fields we might want to hide on this struct to remove the None fields from the serialized response. 

Currently the response to a newly created secret is:
```json
{
	"id": "1d0fd8c8-cb87-4ffe-830d-48d6e11a75dc",
	"guessed": false,
	"clue1": null,
	"clue2": null,
	"clue3": null,
	"guesser": null,
	"secret": null
}
```

Ideally it should only be:
```json
{
	"id": "1d0fd8c8-cb87-4ffe-830d-48d6e11a75dc",
	"guessed": false
}
```

Only adding the extra fields if needed

The code should look something like:
```rust
#[serde(skip_serializing_if = "Option::is_none")]
pub clue1: Option<String>,
```